service/service_strong_types.hpp

```diff
@@ -9,7 +9,7 @@
 
 namespace raven
 {
-  using config_id_st = st::type<int, struct config_id_tag, st::arithmetic>;
+  using config_id_st = st::type<std::size_t, struct config_id_tag, st::arithmetic>;
```

Comment:
I think we can safely assume the client will load less than 2^32 configs. You could even strengthen that assumption to 2^16 or even 2^8 (that last one may be too far though).